File: python_files/ex12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    # Sobrecarga do operador +
    def __add__(self, outro_produto):
        return self.preco + outro_produto.preco

    def __str__(self):
        return f"Produto: {self.nome}, Preço: {self.preco:.2f}"

produto1 = Produto("Teclado", 100.0)
produto2 = Produto("Mouse", 50.0)

print(produto1)
print(produto2)
preco_total = produto1 + produto2
print(f"Preço total: {preco_total}")

File: go_files/ex12.go
package main

import "fmt"

type Produto struct {
	nome string
	preco float64
}

// Método de conveniência para somar o preço dos produtos
func (p Produto) SomarPreco(p2 Produto) float64 {
	return p.preco + p2.preco
}

func (p Produto) String() string {
    return fmt.Sprintf("Produto: %s, Preço: %.2f", p.nome, p.preco)
}

func main() {
    produto1 := Produto{nome: "Teclado", preco: 100.0}
    produto2 := Produto{nome: "Mouse", preco: 50.0}

    fmt.Println(produto1)
    fmt.Println(produto2)
    precoTotal := produto1.SomarPreco(produto2)
    fmt.Printf("Preço total: %.2f\n", precoTotal)
}